nearcore/src/state_sync.rs

```diff
@@ -304,6 +328,249 @@ async fn state_sync_dump(
     tracing::debug!(target: "state_sync_dump", shard_id, "Stopped state dump thread");
 }
 
+async fn get_missing_state_parts_for_epoch(
+    shard_id: ShardId,
+    chain_id: &String,
+    epoch_id: &EpochId,
+    epoch_height: u64,
+    total_parts: u64,
+    external: &ExternalConnection,
+) -> Result<Vec<u64>, anyhow::Error> {
+    let s3_directory =
+        external_storage_location_directory_multi_node(chain_id, epoch_id, epoch_height, shard_id);
+    let file_names = external.list_state_parts(shard_id, &s3_directory).await?;
+    let mut res = vec![];
+    if !file_names.is_empty() {
+        let mut total_parts: u64 = 0;
+        let mut existing_nums = HashSet::new();
+        for name in file_names {
+            let splitted: Vec<_> = name.split("_").collect();
+            let part_id = splitted.get(2).unwrap().to_string().parse::<u64>()?;
+            if total_parts == 0 {
```

Comment:
Why would `total_parts` be `0`?
If there is some special case to represent not-initialized, then please consider using `None` and `Some()` otherwise.